packages/admin/cms-admin/src/userPermissions/user/UserPage.tsx

```diff
@@ -107,10 +82,30 @@ export const UserPage = ({ userId }: { userId: string }) => {
                     <SupportText>{data.user.email}</SupportText>
                 </ToolbarTitleItem>
                 <ToolbarFillSpace />
-                {isAllowed("impersonation") && (
-                    <CardContent>
-                        <ImpersonationButton userId={userId} />
-                    </CardContent>
+                {/* TODO: remove this condition once more actions are added to that menu */}
+                {userId !== currentUser.id && isAllowed("impersonation") && (
+                    <CrudMoreActionsMenu
+                        slotProps={{
+                            group: { groupTitle: null },
+                        }}
+                        overallActions={[
+                            isAllowed("impersonation")
```

Comment:
Here a example how you can improve your conditions and make your code more readabile

`const impersonateButton = ({ userId, currentUser, startImpersonation, isAllowed }) => {
    // Destructure currentUser.id for cleaner usage
    const { id: currentUserId } = currentUser;

    // If impersonation is not allowed, return null early
    if (!isAllowed("impersonation")) return null;

    // Create memoized onClick handler
    const handleImpersonation = useCallback(() => startImpersonation(userId), [userId, startImpersonation]);

    return {
        label: <FormattedMessage id="comet.impersonate" defaultMessage="Impersonate" />,
        icon: <ImpersonateUser />,
        disabled: userId === currentUserId, // Use destructured currentUserId
        onClick: handleImpersonation, // Use memoized handler
    };
};
`